src/makeInfinityAdder.js

```diff
@@ -4,7 +4,32 @@
  * @return {function}
  */
 function makeInfinityAdder() {
-  // write code here
+  let totalSum = 0;
+  let currentSum = 0;
+
+  function adder(num) {
+    if (num === undefined) {
+      const result = totalSum + currentSum;
+
+      totalSum = 0;
+      currentSum = 0;
+
+      return result;
+    } else {
+      currentSum += num;
+
+      return adder;
+    }
+  }
+
+  adder.toString = () => {
+    totalSum += currentSum;
+    currentSum = 0;
+
+    return totalSum;
+  };
```

Comment:
The `toString` method is not required for this task. The function should return the sum when called without arguments, not when its string representation is requested. This method can be removed.